### ISSUE TITLE
RAM filled fastly while `ranger` tried to identify `djvu` "orientation".

### ISSUE THREAD

**Issue by firmart** [ranger/ranger#1815]
#### Runtime Environment

- Operating system and version: Ubuntu 19.04 x86_64
- Terminal emulator and version: XTerm(330) and termit 3.0.0
- ranger version: ranger 1.9.2
- Python version: 3.7.3 (default, Oct  7 2019, 12:56:13) [GCC 8.3.0]
- Locale: en_US.UTF-8

#### Current Behavior

I was on a 5MB `.djvu`  and my 16GB of RAM was fulled within 3 seconds. More specifically, I've seen in `htop` (before my computer gets frozen) that the command `identify -format '%[EXIF:Orientation]\n' --  file.djvu` was run (which comes from `scope.sh`). I guess there is an issue when `ranger` is trying to identify the orientation of the `djvu` file (MIME: `image/vnd.djvu`). The problem probably comes from `djvu` mime which is categorized as `image`.

**Comment by toonn** [comment]
Is this reproducible with other djvu files? (Preferably sample djvu files.)

**Comment by firmart** [comment]
I have tried to create a one-page `.djvu` file, but nothing happens with it. However, with `.djvu` of 100+ pages, the issue is reproduced (didn't try intermediates). The attached sample `.djvu` file is created by the following `sample.tex` file :  
```latex
\documentclass{article}
\usepackage{lipsum}
\begin{document}
\Huge
\lipsum[1-150]
\end{document}
```
with the command `pdflatex sample.tex && pdf2djvu sample.pdf > sample.djvu`.
[sample-djvu.zip (500KB)](https://github.com/ranger/ranger/files/4109593/sample-djvu.zip)

**Comment by firmart** [comment]
An easy way to solve the problem is to select `image` mime type only supported by `identify` instead to quantify universally all `image` type which includes `.djvu`.

That's it, instead of this,

https://github.com/ranger/ranger/blob/3965e3eeafac8ce44670cfdc25ccc7b5997a3366/ranger/data/scope.sh#L142-L145
do this for instance:
```bash
 image/png | image/jpeg) 
     local orientation 
     orientation="$( identify -format '%[EXIF:Orientation]\n' -- "${FILE_PATH}" )" 
     ## If orientation data is present and the image actually
```

**Comment by toonn** [comment]
Ok, so we've narrowed it down to multiple versions of ranger with multiple versions of imagemagick on multiple distros with multiple file types...

Maybe we can put an alarm on the call to identify or something?

**Comment by firmart** [comment]
Hi @toonn, 

I got an idea which seems to work: using `ulimit` to bound temporarily  `identify` resources. 
It's rather a hack, but I think it's better than nothing. I found in other software similar issue when they trying to handle `djvu` files. If we want to understand why we have to deal with `imagemagick`...

**Comment by glyh** [comment]
Is this fixed? I still run into this frequently.

**Comment by glyh** [comment]
This is fixed, I check the source code, they handle djvu file separately, so we should close this. If you're using arch linux, please use `ranger-git` instead of `ranger`.

**Comment by i007c** [comment]
with prlimit i fixed this issue
```
orientation="$( prlimit -v=200000000 identify -format '%[EXIF:Orientation]\n' -- "${FILE_PATH}" )"
```
i just gave it 200 meg of ram 
it fine with normal images
and logs an error message with masive images

**Comment by toonn** [comment]
This should be fixed or rather worked-around by #2743.
